uaclient/contract.py

```diff
@@ -497,3 +566,38 @@ def get_contract_information(cfg: UAConfig, token: str) -> Dict[str, Any]:
     """Query contract information for a specific token"""
     client = UAContractClient(cfg)
     return client.request_contract_information(token)
+
+
+def is_contract_changed(cfg: UAConfig) -> bool:
+    orig_token = cfg.machine_token
+    orig_entitlements = cfg.entitlements
+    machine_token = orig_token.get("machineToken", "")
+    contract_id = (
+        orig_token.get("machineTokenInfo", {})
+        .get("contractInfo", {})
+        .get("id", "")
+    )
+    contract_client = UAContractClient(cfg)
+    resp = contract_client.get_updated_contract_info(
+        machine_token, contract_id
+    )
+    resp_expiry = (
+        resp.get("machineTokenInfo", {})
+        .get("contractInfo", {})
+        .get("effectiveTo", None)
+    )
+    new_expiry = (
+        util.parse_rfc3339_date(resp_expiry)
```

Comment:
I think we should double check if the code will not break if `resp_expiry` is `None`. Looking at `parse_rfc3339_date`, it seems that it will, so it is worth double checking

Reply:
This is part of a one-line if statement python ternary thing (personally not a fan of this python-ism).
So in the next lines you can see that the function ill only be called `if resp_expiry`

Reply:
ohhh my mistake here, missed the ternary operation.